### DISCUSSION TITLE
LazyVim with QT6 possible?

### DISCUSSION THREAD

**Discussion by devhaocui02** [LazyVim/LazyVim#3535] [Q&A]
I was wondering, been a fan of LazyVim and NeoVim in general and was wondering if I could make a development environment for making and running QT6 without accessing QT's IDE?

**Comment by Nicknamely** [comment]
Yes, it is possible. I’m currently trying to get the LSP working on my end too (I’m on Arch, by the way). If you’re on Linux, you’ll need to install the Qt tools you need, along with the qt-language-server.

```bash
sudo pacman -S qt6 # This will display all available packages from the qt6 group - install what you need.
```

Look for qt6-language-server.

Then add the LSP in your LSP config like this:

```lua
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      -- QML support
      qmlls = {
        cmd = { "qmlls" },
        filetypes = { "qml", "qmljs" },
      },
      -- --
    },
  },
}
```

And for syntax highlighting, make sure you have it in `ensure_installed`:
```lua
return {
  {
    "nvim-treesitter/nvim-treesitter",
    opts = function(_, opts)
      vim.list_extend(opts.ensure_installed, {
        -- "typescript",
        "css",
        "scss",
        "latex",
        "norg",
        "svelte",
        "typst",
        "vue",
        "qmljs", -- here
      })
    end,
  },
}
```

That’s all I’ve done so far, but I can’t get the LSP working yet - it returns:
```lua
/usr/share/nvim/runtime/lua/vim/lsp/_transport.lua:68: Spawning language server with cmd: `{ "qmlls" }` failed. The language server is either not installed, missing from PATH, or not executable.
```

Edit: In your LSP configuration, you need to use the cmd qmlls6 instead

**Reply by SouvikDuttaChowdhury** [reply]
The cmd for qmlls in arch is either qmlls6 or qmlls5 depending on the version installed

**Comment by Nicknamely** [comment]
I am reviving this post since having an extra for qt dev  would be great.
